### ISSUE TITLE
Links as variables

### ISSUE THREAD

**Issue by kouloumos** [chaincodelabs/onboarding-to-bitcoin-core#21]
This document as well as https://github.com/chaincodelabs/bitcoin-core-onboarding are based on a specific reference of the codebase. Throughout the onboarding documentation there are links that are referenced more than once. I was thinking that moving all the links to variables would remove duplicate links but more importantly would make easier any additions to the documentation along with easier upgradability in case of future migration to a newer version of the codebase.

I've started playing along with the idea, initially for [Regions](https://github.com/chaincodelabs/bitcoin-core-onboarding/blob/main/1.1_regions.asciidoc) but as it is referencing the [same commit](https://github.com/bitcoin/bitcoin/commit/4b5659c6b115315c9fd2902b4edd4b960a5e066e), is easy to just have the same source of truth for everything. My idea is something like
```
:commit: 4b5659c6b115315c9fd2902b4edd4b960a5e066e
:link: https://github.com/bitcoin/bitcoin/blob/{commit}/src

// files
:net_processing_h: {link}/net_processing.h
:net_processing_cpp: {link}/net_processing.cpp
:validation_cpp: {link}/validation.cpp
:validation_h: {link}/validation.h
...
// net_processing.{h,cpp}
:PeerManagerImpl: {net_processing_cpp}#L227
:m_chainman: {net_processing_cpp}#L335
...
// validation.{h,cpp}
:ConnectBlock: {validation_cpp}#L1802
:ConnectTip: {validation_cpp}#L2460
:ActivateBestChainStep: {validation_cpp}#L2596
```

which could be imported on every doc with
```
include::links.adoc[]
```

The only problem with this  is that including asciidoc files [is not working on GitHub](https://github.com/github/markup/issues/1095) which is an issue for people reading the documentation from within GitHub and not from the website. To be able to work for GitHub it would need all the variables to be presented in each file, which complicates the maintenance process. A solution to that could be adding [asciidoctor-reducer](https://github.com/asciidoctor/asciidoctor-reducer) as a pre-commit hook.

I wanted to get some feedback to understand if others see value on this and if the GitHub limitation is consider a forbidding factor.

**Comment by willcl-ark** [comment]
@kouloumos I agree that this would have been a nice approach, in fact, this is how I had it back around https://github.com/chaincodelabs/onboarding-to-bitcoin-core/commit/57a88097f642abaa80da460bf33f60ffa19c488b !

I then reverted to this format, as, although it makes the unrendered text a bit more "messy" to work with, contributors would be able to visit links directly from within their editors without having to append (shift+click to open in Alacritty for example).

I also got frustrated with how few of the asciidoc features github enabled, and it felt very freeing to be able to use the full featureset again once I dropped the asciidoctor-reducer precommit hook step. In addition to this, rebasing (which is probably unnecessary tbh) became a real nightmare with the reduced document, as it always needed manual adjustment between diffs.

Overall I felt that it was not going to make sense either for upgrading, as if we, for example, wanted to change from our current commit to `v23.0`, every link's lines would need to be manually checked and updated anyway, so the link variable would not help us much here either.

**Comment by kouloumos** [comment]
I didn't persuade this further mainly because of Github rendering and your comment

> contributors would be able to visit links directly from within their editors without having to append

Apart from that, I still think that variables would make future upgrades easier by separating changes in context and changes in links. But still, if that's the only benefit a parse/find-and-replace script could be written when we reach that point.
